src/threading.c

```diff
@@ -334,6 +334,41 @@ static jl_value_t *ti_run_fun(jl_callptr_t fptr, jl_method_instance_t *mfunc,
 jl_mutex_t codegen_lock;
 jl_mutex_t typecache_lock;
 
+// Init RCU global counter to zero
+int rcu_gp_ctr;
+
+void rcu_read_lock(jl_ptls_t ptls)
+{
+    int tmp = rcu_gp_ctr;
+    jl_atomic_fetch_add(&tmp, 1);
```

Comment:
It's never useful to have an atomic operation on a variable that will only be access by one thread....